src/RiveQtQuickItem/renderer/riveqtrhirenderer.cpp

```diff
@@ -681,31 +678,32 @@ TextureTargetNode *RiveQtRhiRenderer::getRiveDrawTargetNode()
 
 void RiveQtRhiRenderer::setProjectionMatrix(const QMatrix4x4 *projectionMatrix, const QMatrix4x4 *modelMatrix)
 {
-    auto aspectX = m_item->width() / (m_artboardSize.width());
-    auto aspectY = m_item->height() / (m_artboardSize.height());
+    const auto item2artboardScaleX = m_item->width() / m_artboardSize.width();

```

Comment:
The `RiveQSGBaseNode` does now have two members

```
QPointF m_topLeftRivePosition;
float m_scaleFactor { 1.0f };
```

those are used to tell the quickItem (hit-test) where the rendering takes place - I think the information here should also update those members - or calculate the RenderSettings depending elements in the Node (not in the renderer) - and just set the result to the renderer - this way it can be done once and used for QPainter and OpenGL Rendering as well

Reply:
m_scaleFactor will not work, when the item is stretched and there are different x and y scalings, or am I mistaken?

I can put the information in the render settings later - but I'd say that this MR has a different purpose

Reply:
true, ScaleX and ScaleY, well since you do changes to it in the PR lets just put it right here. Change m_scaleFactor to m_scaleFactorX and m_scaleFactorY and fix it also for Qt5 Version.

Reply:
Changed it. It works in OpenGl (Qt5), Software and RHI Rendernodes.
The OpenGl node does no scissoring of the Artboard but on the whole item, so the behavior is the same as in the RHI rendernode.